libraries/botframework-config/src/botConfiguration.ts

```diff
@@ -189,73 +265,168 @@ export class BotConfiguration extends BotConfigurationBase {
         }
     }
 
-    // save the config file to specificed botpath
-    public async saveAs(botpath: string, secret?: string): Promise<void> {
-        if (!botpath) {
-            throw new Error(`missing path`);
-        }
-
-        this._savePrep(secret);
-
-        const hasSecret: boolean = !!this.secretKey;
-
-        if (hasSecret) {
-            this.encrypt(secret);
-        }
-        await fsx.writeJson(botpath, this.toJSON(), { spaces: 4 });
-
-        if (hasSecret) {
-            this.decrypt(secret);
-        }
-    }
-
-    // save the config file to specificed botpath
-    public saveAsSync(botpath: string, secret?: string): void {
-        this._savePrep(secret);
+    // export the services from the bot file as resource files and recipe file
```

Comment:
Consider using more rigorous language in code comments on public methods on public classes. These words often end up in public facing documentation, and having complete sentences with proper capitalization is important.